packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -73,8 +98,110 @@ class CupertinoNavigationBar extends StatelessWidget implements PreferredSizeWid
   /// True if the nav bar's background color has no transparency.
   bool get opaque => backgroundColor.alpha == 0xFF;
 
+  /// Use iOS 11 style large title navigation bars.
+  ///
+  /// When true, the navigation bar will split into 2 sections. The static
+  /// top 44px section will be wrapped in a SliverPersistentHeader and a
+  /// second scrollable section behind it will show and replace the `middle`
+  /// text in a larger font when scrolled down.
+  ///
+  /// Navigation bars with large titles must be used in a sliver group such
+  /// as [CustomScrollView].
+  final bool largeTitle;
+
+  @override
+  Size get preferredSize => const Size.fromHeight(_kNavBarPersistentHeight);
+
+  @override
+  Widget build(BuildContext context) {
+    assert(
+      !largeTitle || middle is Text,
+      "largeTitle mode is only possible when 'middle' is a Text widget",
```

Comment:
we should never require a particular kind of widget. that's an anti-pattern.

Reply:
Ah ok. Understood. I thought there was prior art (like CustomScrollView doesn't work if the children Widget don't do certain things). 

I'll break up the CupertinoNavigationBar into 2 versions, one for static and one with slivers like material. Then the `largeTitle` can just be a Text instead of a bool. The iOS APIs allow just a string anyway.